frontend/src/actions/chat.ts

```diff
@@ -1,67 +1,28 @@
-'use server'
+"use server"
 
-import {CoursesService} from '@/client'
-import {get} from '@/utils'
+import { ChatMessageUI } from "@/client/client/types.gen";
+import { ChatService } from "@/client/sdk.gen";
+import { ChatMessage } from "@/client/zod.gen";
+import { get } from '@/utils'
 
-export interface ChatMessage {
-  id: string
-  is_system: boolean
-  message: string
-  author?: string
-  avatar?: string
-}
-
-const mockApiRequest = (): Promise<{id: string, message: string}> => {
-  return new Promise((resolve, reject) => {
-    setTimeout(() => {
-      const data = { message: "This is system message", id: Date.now().toString() };
-      resolve(data);
-    }, 2000);
-  });
-}
-
-export async function sendChat({ id, message }: {id?: string, message: string}): Promise<ChatMessage | undefined> {
-  try {
-    // change the api call when chat api is ready
-    const response = await mockApiRequest();
-    return {
-      id: response.id,
-      message: response.message,
-      is_system: true
-    };
-  } catch (error) {
-    console.error(error)
-    const errorMsg = get(
-      error as Record<string, never>,
-      'detail',
-      'API request failed',
-    )
-
-    throw new Error(errorMsg)
-  }
-}
-
-const mockHistoryApiRequest = (): Promise<{messages: ChatMessage[]}> => {
-  return new Promise((resolve, reject) => {
-    setTimeout(() => {
-      const data = { messages: []};
-      resolve(data);
-    }, 2000);
-  });
-}
-
-export async function getHistory({ cursor }: {cursor?: string}): Promise<ChatMessage[]> {
+export async function getChatHistory(courseId: string): Promise<ChatMessageUI[]> {
   try {
-    const response = await mockHistoryApiRequest()
-    return [];
+    const res = await ChatService.getApiV1ChatHistory({path: { course_id: courseId }});
```

Comment:
I've standardized the way errors and data fetching are handled to leverage TypeScript types. See https://github.com/A3copilotprogram/plg_6/blob/0eedde74e766987d53d8b3e756cbabc0851ff7da/frontend/src/actions/courses.ts